models/hh_psc_alpha_clopath.h

```diff
@@ -0,0 +1,440 @@
+/*
+ *  hh_psc_alpha_clopath.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef HH_PSC_ALPHA_CLOPATH_H
+#define HH_PSC_ALPHA_CLOPATH_H
+
+// Generated includes:
+#include "config.h"
+
+#ifdef HAVE_GSL
+
+// External includes:
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_odeiv.h>
+#include <gsl/gsl_sf_exp.h>
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "recordables_map.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+/**
+ * Function computing right-hand side of ODE for GSL solver.
+ * @note Must be declared here so we can befriend it in class.
+ * @note Must have C-linkage for passing to GSL. Internally, it is
+ *       a first-class C++ function, but cannot be a member function
+ *       because of the C-linkage.
+ * @note No point in declaring it inline, since it is called
+ *       through a function pointer.
+ * @param void* Pointer to model neuron instance.
+ */
+extern "C" int
+hh_psc_alpha_clopath_dynamics( double, const double*, double*, void* );
+
+/** @BeginDocumentation
+Name: hh_psc_alpha_clopath - Hodgkin-Huxley neuron model.
```

Comment:
We found an example implementation on (I think) ModelDB that uses the Clopath rule with a Hodgkin-Huxley model. Thus we thought that this would be a nice addition and extend the usability of the synapse in NEST.

The model is 95% identical to `hh_psc_alpha`, however includes an unnecessary overhead for simulations without the clopath rule, which is why we created a new model. I will look for a unit test of `hh_psc_alpha` that can be adapted with an additional test of the clopath rule functionality

Reply:
Added the missing test with f821cec